src/components/IncompatibleEditorDevice.tsx

```diff
@@ -0,0 +1,122 @@
+import {
+  Button,
+  HStack,
+  Heading,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalOverlay,
+  Text,
+  VStack,
+} from "@chakra-ui/react";
+import { FormattedMessage } from "react-intl";
+import { useProject } from "../hooks/project-hooks";
+import { ReactNode } from "react";
+
+interface UnsupportedEditorDeviceProps {
+  isOpen: boolean;
+  onClose: () => void;
+  onNext?: () => void;
+  onBack?: () => void;
+  stage: "openEditor" | "flashDevice";
+}
+
+const UnsupportedEditorDevice = ({
+  isOpen,
+  onClose,
+  onNext,
+  onBack,
+  stage,
+}: UnsupportedEditorDeviceProps) => {
+  const { saveHex } = useProject();
+  return (
+    <Modal
+      closeOnOverlayClick={false}
+      motionPreset="none"
+      isOpen={isOpen}
+      onClose={onClose}
+      size="3xl"
+      isCentered
+    >
+      <ModalOverlay>
+        <ModalContent p={8}>
+          <ModalBody>
+            <ModalCloseButton />
+            <VStack width="100%" alignItems="left" gap={5}>
+              <Heading as="h2" fontSize="xl" fontWeight="bold">
+                <FormattedMessage id="incompatible-device-heading" />
+              </Heading>
+              <VStack gap={5} align="stretch">
+                <Text>
+                  <FormattedMessage id="incompatible-device-subtitle" />
+                </Text>
+                {stage === "openEditor" ? (
+                  <>
+                    <Text>
+                      <FormattedMessage id="incompatible-device-body-1" />
+                    </Text>
+                    <Text>
+                      <FormattedMessage
+                        id="incompatible-device-body-2"
+                        values={{
+                          link: (chunks: ReactNode) => (
+                            <Text
+                              as="button"
+                              color="brand.600"
+                              onClick={() => saveHex()}
+                            >
+                              {chunks}
+                            </Text>
+                          ),
+                        }}
+                      />
+                    </Text>
+                  </>
+                ) : (
+                  <Text>
+                    <FormattedMessage
+                      id="incompatible-device-body-alt"
+                      values={{
+                        link: (chunks: ReactNode) => (
+                          <Text
```

Comment:
I think these are Button variant=link elsewhere. Would that work here too?

Reply:
Done here. Updated another instance as well [a0adffe](https://github.com/microbit-foundation/ml-trainer/pull/358/commits/a0adffe6268b08a55eaeee30f6a562b51dfb8972)